data-pipeline-ffi/src/trace_exporter.rs

```diff
@@ -89,7 +89,7 @@ pub unsafe extern "C" fn ddog_trace_exporter_free(handle: Box<TraceExporter>) {
 /// * `trace_count` - The number of traces to send to the Datadog Agent.
 #[no_mangle]
 pub unsafe extern "C" fn ddog_trace_exporter_send(
-    handle: &TraceExporter,
+    handle: &mut TraceExporter,
```

Comment:
We can avoid making the entire TraceExporter mutable by leveraging something like [RefCell](https://doc.rust-lang.org/std/cell/struct.RefCell.html) to achieve [interior mutability](https://doc.rust-lang.org/book/ch15-05-interior-mutability.html). 

[This commit compiles](https://github.com/DataDog/libdatadog/commit/16b4bcc0c01b5779e7464aebf4103c6162d2095e).

There is a minor performance hit to using RefCell, although I'm not sure we'd notice it. 

It _feels right_ to me for the TraceExporter to be immutable, but I'm not sure that's sufficient justification? I'm curious what other people think.